### PR TITLE
refactor: preinsertunique

### PR DESCRIPTION
**Pull request by huby2358**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [x] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

### CODE REVIEW

**Review by ouyuanning — APPROVED** [review]

**Review by badboynt1 — APPROVED** [review]

File: pkg/sql/colexec/preinsertsecondaryindex/preinsertsecondaryindex.go [hunk 1]
```diff
@@ -93,7 +93,8 @@ func (preInsertSecIdx *PreInsertSecIdx) Call(proc *process.Process) (vm.CallResu
 	// colCount >= 2 is more common.
 	if colCount == 1 {
 		pos := secondaryColumnPos[indexColPos]
-		vec, bitMap, err = util.CompactSingleIndexCol(inputBat.Vecs[pos], proc)
+		vec := proc.GetVector(*inputBat.Vecs[pos].GetType())
```
Comment (fengttt) [inline pkg/sql/colexec/preinsertsecondaryindex/preinsertsecondaryindex.go:96]: You cannot use this

File: pkg/sql/colexec/preinsertsecondaryindex/preinsertsecondaryindex.go [hunk 2]
```diff
@@ -110,7 +111,8 @@ func (preInsertSecIdx *PreInsertSecIdx) Call(proc *process.Process) (vm.CallResu
 	preInsertSecIdx.ctr.buf.SetVector(indexColPos, vec)
 	preInsertSecIdx.ctr.buf.SetRowCount(vec.Length())
 
-	vec, err = util.CompactPrimaryCol(inputBat.Vecs[pkPos], bitMap, proc)
+	vec = proc.GetVector(*inputBat.Vecs[pkPos].GetType())
```
Comment (fengttt) [inline pkg/sql/colexec/preinsertsecondaryindex/preinsertsecondaryindex.go:114]: This too.    If you really need a vector, not a batch, put the vector in your ctr.   Do not lose it.   Make a comment in types.go Use it, reuse it, and free it at the right time.

**Review by fengttt — CHANGES_REQUESTED** [review]
get rid of the pointer of ctr.   Useless pointer.

You still have PutBatch, which is completely wrong.

**Comment by ouyuanning** [comment]
> get rid of the pointer of ctr. Useless pointer.
> 
> You still have PutBatch, which is completely wrong.

seems done

**Pull request merged**